src/Serilog.Sinks.Elasticsearch/Sinks/ElasticSearch/ElasticsearchSinkOptions.cs

```diff
@@ -41,6 +41,18 @@ public class ElasticsearchSinkOptions
         /// </summary>
         public string TemplateName { get; set; }
 
+        /// <summary>
+        /// When using the <see cref="AutoRegisterTemplate"/> feature, this allows you to override the default template content.
+        /// If not provided, a default template that is optimized to deal with Serilog events is used.
+        /// </summary>
+        public Func<string> GetTemplateContent { get; set; }
```

Comment:
It might be more useful to return `object` here, the implementer could return a string or anonymous object in that case.